### ISSUE TITLE
"type" attribute not fully implemented in the button.ts component

### ISSUE THREAD

**Issue by mraumer-icrossing** [material-components/material-web#5658]
### What is affected?

Component

### Description

When I pass in a `type` to the md-button element I expect that type to be passed to the rendered button. However, in the button.ts code the `type` [property is defined](https://github.com/material-components/material-web/blob/main/button/internal/button.ts#L85) but never used in the `renderButton` method:
```javascript
private renderButton() {
    // Needed for closure conformance
    const {ariaLabel, ariaHasPopup, ariaExpanded} = this as ARIAMixinStrict;
    return html`<button
      id="button"
      class="button"
      ?disabled=${this.disabled}
      aria-label="${ariaLabel || nothing}"
      aria-haspopup="${ariaHasPopup || nothing}"
      aria-expanded="${ariaExpanded || nothing}">
      ${this.renderContent()}
    </button>`;
  }
```

Also, the docs state 

> May be "text", "reset", or "submit" (default).

but the list of supported values is actually `button|submit|reset|dialog` and `text` is unsupported as a type. 

### Reproduction

[from the code here](https://github.com/material-components/material-web/blob/main/button/internal/button.ts#L152C3-L164C4): 
```javascript
private renderButton() {
    // Needed for closure conformance
    const {ariaLabel, ariaHasPopup, ariaExpanded} = this as ARIAMixinStrict;
    return html`<button
      id="button"
      class="button"
      ?disabled=${this.disabled}
      aria-label="${ariaLabel || nothing}"
      aria-haspopup="${ariaHasPopup || nothing}"
      aria-expanded="${ariaExpanded || nothing}">
      ${this.renderContent()}
    </button>`;
  }
```

### Workaround

I have not found any workaround

### Is this a regression?

No or unsure. This never worked, or I haven't tried before.

### Affected versions

Failing in 1.5.0

### Browser/OS/Node environment

All. This is a code issue.

**Comment by asyncLiz** [comment]
The `type` attribute does not get passed to the inner `<button>`, it's used by the host element [to control form submission](https://github.com/material-components/material-web/blob/b75bd2ffb8314b6e5f826ab9d39d5b6e4b7d8850/internal/controller/form-submitter.ts#L82-L90).

While it mimics the behavior of [`HTMLButtonElement.type`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLButtonElement/type), it isn't passed to the inner `<button>` because the type attribute doesn't work across shadow root boundaries.

```html
<form>
  <custom-button>
    #shadow-root
      <button type="reset">Doesn't work!</button>
  </custom-button>
</form>
```

For the docs, they should be updated soon, we just fixed it for https://github.com/material-components/material-web/issues/5650.

**Comment by mraumer-icrossing** [comment]
> The `type` attribute does not get passed to the inner `<button>`, it's used by the host element [to control form submission](https://github.com/material-components/material-web/blob/b75bd2ffb8314b6e5f826ab9d39d5b6e4b7d8850/internal/controller/form-submitter.ts#L82-L90).
> 
> While it mimics the behavior of [`HTMLButtonElement.type`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLButtonElement/type), it isn't passed to the inner `<button>` because the type attribute doesn't work across shadow root boundaries.
> 
> ```
> <form>
>   <custom-button>
>     #shadow-root
>       <button type="reset">Doesn't work!</button>
>   </custom-button>
> </form>
> ```
> 
> For the docs, they should be updated soon, we just fixed it for #5650.

Understood, thank you!
